### ISSUE TITLE
controller gets panic when using configmap-errorfiles

### ISSUE THREAD

**Issue by bsctl** [haproxytech/kubernetes-ingress#394]
We're trying to use the `--configmap-errorfiles` option to customise the error message as for example: `403 Forbidden` code.
The controller gets panic.

Helm values

```yaml
  ## Additional command line arguments to pass to Controller
  ## ref: https://github.com/haproxytech/kubernetes-ingress/blob/master/documentation/controller.md
  extraArgs:
  - --configmap-errorfiles=ingress-system/errorfiles-config 
  #  - --namespace-whitelist=default
  #  - --namespace-whitelist=namespace1
  #  - --namespace-blacklist=namespace2

  ## Custom configuration for Controller
  ## ref: https://github.com/haproxytech/kubernetes-ingress/tree/master/documentation
  config:
    timeout-connect: "250ms"
    rate-limit: "ON"
    rate-limit-expire: "1m"
    rate-limit-interval: "10s"
    rate-limit-size: "100k"
    frontend-config-snippet: |
      http-response set-header X-XSS-Protection "1; mode=block"
      http-response set-header X-Content-Type-Options "nosniff"
      http-response set-header X-Frame-Options "deny"
      http-response set-header Referrer-Policy "no-referrer-when-downgrade"
      http-response set-header Strict-Transport-Security "max-age=31536000;includeSubDomains;preload"
...
```

Config Map

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    app.kubernetes.io/instance: ingress
    app.kubernetes.io/name: kubernetes-ingress
  name: errorfiles-config
  namespace: ingress-system
data:
  503: |-
    HTTP/1.0 503 Service Unavailable
    Cache-Control: no-cache
    Connection: close
    Content-Type: text/html

    <html><body><h1>Oops, that's embarrassing!</h1>
    There are no servers available to handle your request.
    </body></html>
```

Logs:
```
 _   _    _    ____                                            
| | | |  / \  |  _ \ _ __ _____  ___   _                       
| |_| | / _ \ | |_) | '__/ _ \ \/ / | | |                      
|  _  |/ ___ \|  __/| | | (_) >  <| |_| |                      
|_| |_/_/   \_\_|   |_|  \___/_/\_\\__, |                      
 _  __     _                       |___/             ___ ____  
| |/ /   _| |__   ___ _ __ _ __   ___| |_ ___  ___  |_ _/ ___| 
| ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|  | | |      
| . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \  | | |___ 
|_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/ |___\____| 
                                                                                                                         

2021/11/15 21:31:42 HAProxy Ingress Controller v1.7.1 04bbbc0
2021/11/15 21:31:42 Build from: https://github.com/haproxytech/kubernetes-ingress
2021/11/15 21:31:42 Build date: 2021-11-05T11:02:16

2021/11/15 21:31:42 ConfigMap: ingress-system/ingress-kubernetes-ingress
2021/11/15 21:31:42 Ingress class: infra
2021/11/15 21:31:42 Empty Ingress class: false
2021/11/15 21:31:42 Publish service: 
2021/11/15 21:31:42 Default backend service: ingress-system/ingress-kubernetes-ingress-default-backend
2021/11/15 21:31:42 Default ssl certificate: ingress-system/ingress-certs
2021/11/15 21:31:42 Frontend HTTP listening on: 0.0.0.0:80
2021/11/15 21:31:42 Frontend HTTPS listening on: 0.0.0.0:443
2021/11/15 21:31:42 Errofile pages provided in 'ingress-system/errorfiles-config'
2021/11/15 21:31:42 DEBUG   Kubernetes Informers resync period: 10m0s
2021/11/15 21:31:42 Controller sync period: 5s

2021/11/15 21:31:42 Running on infra01
2021/11/15 21:31:42 haproxy.go:36 Running with HAProxy version 2.4.8-d1f8d41 2021/11/03 - https://haproxy.org/
2021/11/15 21:31:42 haproxy.go:50 Starting HAProxy with /etc/haproxy/haproxy.cfg 
2021/11/15 21:31:42 controller.go:116 Running on Kubernetes version: v1.20.6 linux/amd64
2021/11/15 21:31:42 DEBUG   crmanager.go:70 Custom API core.haproxy.org available
2021/11/15 21:31:42 INFO    crmanager.go:75 Global CR defined in API core.haproxy.org
2021/11/15 21:31:42 DEBUG   crmanager.go:70 Custom API core.haproxy.org available
2021/11/15 21:31:42 INFO    crmanager.go:75 Defaults CR defined in API core.haproxy.org
2021/11/15 21:31:42 DEBUG   crmanager.go:70 Custom API core.haproxy.org available
2021/11/15 21:31:42 INFO    crmanager.go:75 Backend CR defined in API core.haproxy.org
2021/11/15 21:31:42 DEBUG   monitor.go:175 Using discovery.k8s.io/v1beta1 endpointslices
2021/11/15 21:31:42 DEBUG   monitor.go:142 watching ingress resources of apiGroup networking.k8s.io/v1:
2021/11/15 21:31:42 DEBUG   store/events.go:321 configmap 'ingress-system/errorfiles-config' processed
2021/11/15 21:31:42 DEBUG   store/events.go:321 configmap 'ingress-system/ingress-kubernetes-ingress' processed
2021/11/15 21:31:43 DEBUG   monitor.go:81 Executing syncPeriod every 5s
2021/11/15 21:31:48 INFO    monitor.go:229 Auxiliary HAProxy config '/etc/haproxy/haproxy-aux.cfg' updated
2021/11/15 21:31:48 DEBUG   global.go:107 Frontend config-snippet updated: [<nil slice> != [http-response set-header X-XSS-Protection "1; mode=block" http-response set-header X-Content-Type-Options "nosniff" http-response set-header X-Frame-Options "deny" http-response set-header Referrer-Policy "no-referrer-when-downgrade" http-response set-header Strict-Transport-Security "max-age=31536000;includeSubDomains;preload"]]
Reload required
2021/11/15 21:31:48 DEBUG   global.go:138 Defaults config updated: [ConnectTimeout: 250 != 5000]
Reload required
2021/11/15 21:31:48 DEBUG   service/service.go:130 Service 'ingress-system/ingress-kubernetes-ingress-default-backend': new backend 'ingress-system-ingress-kubernetes-ingress-default-backend-http', reload required
2021/11/15 21:31:48 DEBUG   service/endpoints.go:118 Server slots in backend 'ingress-system-ingress-kubernetes-ingress-default-backend-http' scaled to match scale-server-slots value: 10, reload required
2021/11/15 21:31:48 DEBUG   service/service.go:130 Service 'monitoring-system/prometheus-k8s': new backend 'monitoring-system-prometheus-k8s-web', reload required
2021/11/15 21:31:48 DEBUG   service/endpoints.go:118 Server slots in backend 'monitoring-system-prometheus-k8s-web' scaled to match scale-server-slots value: 10, reload required
2021/11/15 21:31:48 DEBUG   service/service.go:130 Service 'auth-system/keycloak-access': new backend 'auth-system-keycloak-access-http', reload required
2021/11/15 21:31:48 DEBUG   service/endpoints.go:118 Server slots in backend 'auth-system-keycloak-access-http' scaled to match scale-server-slots value: 10, reload required
2021/11/15 21:31:48 DEBUG   handler/https.go:158 SSLOffload enabeld, reload required
2021/11/15 21:31:48 DEBUG   certs/main.go:152 Secret 'ingress-system/ingress-certs' was updated, reload required
panic: assignment to entry in nil map

goroutine 37 [running]:
github.com/haproxytech/kubernetes-ingress/controller/handler.files.newFile({{0xc00039e3c0, 0xc000082d00}, 0x0}, {0xc000bc84c8, 0x3}, {0xc0004f8000, 0x91})
        /src/controller/handler/pattern-files.go:89 +0x147
github.com/haproxytech/kubernetes-ingress/controller/handler.ErrorFile.Update({{{0x0, 0xc00046c1b0}, 0x0}, 0x0}, {0x4, 0xc00046c150, 0xc00046c180, {0xc00046c1b0, 0xc00046c1e0}, {0xc0005a8000, ...}}, ...)
        /src/controller/handler/errorfile.go:53 +0x3af
github.com/haproxytech/kubernetes-ingress/controller.(*HAProxyController).updateHAProxy(0xc0002de000)
        /src/controller/controller.go:183 +0x718
github.com/haproxytech/kubernetes-ingress/controller.(*HAProxyController).SyncData(0xc0002de000)
        /src/controller/monitor.go:99 +0x2e6
created by github.com/haproxytech/kubernetes-ingress/controller.(*HAProxyController).monitorChanges
        /src/controller/monitor.go:28 +0x79
2021/11/15 21:31:48 
 _   _    _    ____                                            
| | | |  / \  |  _ \ _ __ _____  ___   _                       
| |_| | / _ \ | |_) | '__/ _ \ \/ / | | |                      
|  _  |/ ___ \|  __/| | | (_) >  <| |_| |                      
|_| |_/_/   \_\_|   |_|  \___/_/\_\\__, |                      
 _  __     _                       |___/             ___ ____  
| |/ /   _| |__   ___ _ __ _ __   ___| |_ ___  ___  |_ _/ ___| 
| ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|  | | |      
| . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \  | | |___ 
|_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/ |___\____| 
                                                                                                                         

2021/11/15 21:31:48 HAProxy Ingress Controller v1.7.1 04bbbc0
2021/11/15 21:31:48 Build from: https://github.com/haproxytech/kubernetes-ingress
2021/11/15 21:31:48 Build date: 2021-11-05T11:02:16
```

**Comment by bsctl** [comment]
From logs, it looks like the issue is related to the `ingress-certs` secret:

```
2021/11/15 21:31:48 DEBUG   certs/main.go:152 Secret 'ingress-system/ingress-certs' was updated, reload required
panic: assignment to entry in nil map

```
However, by just removing the option `--configmap-errorfiles` from the extra arguments, it works regularly.

**Comment by Mo3m3n** [comment]
Thanks for reporting this.
Fix available from `1.7.3`
